exchanges/okx/okx_wrapper.go

```diff
@@ -0,0 +1,1218 @@
+		ok.PrintEnabledPairs()
+	}
+	if !ok.GetEnabledFeatures().AutoPairUpdates {
+		return
+	}
+	err := ok.UpdateTradablePairs(context.TODO(), false)
+	if err != nil {
+		log.Errorf(log.ExchangeSys,
+			"%s failed to update tradable pairs. Err: %s",
+			ok.Name,
+			err)
+	}
+}
+
+// FetchTradablePairs returns a list of the exchanges tradable pairs
+func (ok *Okx) FetchTradablePairs(ctx context.Context, a asset.Item) ([]string, error) {
+	if !ok.SupportsAsset(a) {
+		return nil, fmt.Errorf("asset type of %s is not supported by %s", a, ok.Name)
+	}
+	format, err := ok.GetPairFormat(a, false)
+	if err != nil {
+		return nil, err
+	}
+	pairs := []string{}
+	insts := []Instrument{}
+	var er error
+	switch a {
+	case asset.Spot:
+		insts, er = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeSpot,
+		})
+	case asset.Futures:
+		insts, er = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeFutures,
+		})
+	case asset.PerpetualSwap:
+		insts, er = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeSwap,
+		})
+	case asset.Option:
+		var instsb []Instrument
+		var instsc []Instrument
+		insts, er = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeOption,
+			Underlying:     "BTC-USD",
+		})
+		if er != nil {
+			goto checkErrorAndContinue
+		}
+		instsb, er = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeOption,
+			Underlying:     "ETH-USD",
+		})
+		if er != nil {
+			goto checkErrorAndContinue
+		}
+		instsc, er = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeOption,
+			Underlying:     "SOL-USD",
+		})
+		insts = append(insts, instsb...)
+		insts = append(insts, instsc...)
+	case asset.Margin:
+		insts, er = ok.GetInstruments(ctx, &InstrumentsFetchParams{
+			InstrumentType: okxInstTypeMargin,
+		})
+	}
+checkErrorAndContinue:
+	if er != nil || len(insts) == 0 {
+		return pairs, er
+	}
+	for x := range insts {
+		var pair string
+		switch insts[x].InstrumentType {
+		case asset.Spot:
+			pair = insts[x].BaseCurrency + format.Delimiter + insts[x].QuoteCurrency
+		case asset.Futures, asset.PerpetualSwap, asset.Option:
+			currency, err := currency.NewPairFromString(insts[x].Underlying)
+			if err != nil {
+				continue
+			}
+			pair = currency.Base.String() + format.Delimiter + currency.Quote.String()
+		}
+		pairs = append(pairs, pair)
+	}
+	return pairs, nil
+}
+
+// UpdateTradablePairs updates the exchanges available pairs and stores them in the exchanges config
+func (ok *Okx) UpdateTradablePairs(ctx context.Context, forceUpdate bool) error {
+	assetTypes := ok.GetAssetTypes(false)
+	for i := range assetTypes {
+		if !(assetTypes[i] == asset.Spot || assetTypes[i] == asset.PerpetualSwap || assetTypes[i] == asset.Option || assetTypes[i] == asset.Futures || assetTypes[i] == asset.Margin) {
+			return errInvalidInstrumentType
+		}
```

Comment:
This should not happen as we are getting the asset types that are loaded in set defaults.